db/db_test2.cc

```diff
@@ -2897,6 +2897,94 @@ TEST_F(DBTest2, OptimizeForSmallDB) {
 
 #endif  // ROCKSDB_LITE
 
+TEST_F(DBTest2, IterRaceFlush1) {
+  ASSERT_OK(Put("foo", "v1"));
+
+  ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->LoadDependency(
```

Comment:
Will this interleaving cause empty data to be returned without this fix?

Reply:
This is the test case constructed before the fix: https://gist.github.com/siying/ab5c1eb976f2300d154bf6c84076a861

It's not applicable anymore after the fix as the code is shuffled.